mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/scheduler/SchedulingConstraints.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.core.scheduler;
+
+import io.mantisrx.runtime.MachineDefinition;
+import io.mantisrx.shaded.com.google.common.annotations.VisibleForTesting;
+import io.mantisrx.shaded.com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+/**
+ * A class that represents scheduling constraints. These constraints include the resource constraints, and a map of assignment attributes (e.g. jdkVersion:17 or springBootVersion:3).
```

Comment:
For an instance of SchedulingConstraints in the TEAllocationRequest and the instance used in matching TE (e.g. in TEStateManager), are they going to be the same (equals true) or is there another translation happening (e.g enrich(TEAllocation.SchedulingConstraints))?

Reply:
I'll make sure to add better docstrings in case I end up using this class also to index registered TEs in the TEStateManager.